vscode/src/ViolationCodeActionProvider.ts

```diff
@@ -1,45 +1,89 @@
 import * as vscode from "vscode";
-
+import { ExtensionState } from "./extensionState";
+import { EnhancedIncident } from "@editor-extensions/shared";
+import { Immutable } from "immer";
+import { DiagnosticSource } from "@editor-extensions/shared";
 export class ViolationCodeActionProvider implements vscode.CodeActionProvider {
   static readonly providedCodeActionKinds = [vscode.CodeActionKind.QuickFix];
 
-  provideCodeActions(
+  constructor(private state: ExtensionState) {}
+
+  private findMatchingIncident(
+    diagnostic: vscode.Diagnostic,
+  ): Immutable<EnhancedIncident> | undefined {
+    if (typeof diagnostic.code !== "string") {
+      return undefined;
+    }
+    const index = parseInt(diagnostic.code, 10);
+
+    if (isNaN(index)) {
+      console.error("Invalid index in diagnostic code:", diagnostic.code);
+      return undefined;
+    }
+
+    // Get the incident at the specified index
+    const incidents = this.state.data.enhancedIncidents;
+    if (index < 0 || index >= incidents.length) {
+      console.error(
+        `Index ${index} is out of range for incidents array (length: ${incidents.length})`,
+      );
+      return undefined;
+    }
+
+    const incident = incidents[index];
+
+    return incident;
+  }
+
+  async provideCodeActions(
     document: vscode.TextDocument,
     _range: vscode.Range,
     context: vscode.CodeActionContext,
     _token: vscode.CancellationToken,
-  ): vscode.ProviderResult<vscode.CodeAction[]> {
+  ): Promise<vscode.CodeAction[]> {
     const actions: vscode.CodeAction[] = [];
+    const continueExt = vscode.extensions.getExtension("Continue.continue");
 
-    for (const diagnostic of context.diagnostics) {
-      if (diagnostic.source === "AnalysisEngine") {
-        const action = this.createQuickFix(document, diagnostic);
-        actions.push(action);
-      }
+    // Only process if there are diagnostics
+    if (context.diagnostics.length === 0) {
+      return actions;
     }
 
-    return actions;
-  }
+    // Get the first diagnostic that's from our source
+    const diagnostic = context.diagnostics.find((d) => d.source === DiagnosticSource);
```

Comment:
What I am seeing is the 2 diagnostics do show up in the problems list ok.  The quick actions view is where some potential confusion may lie if a user wants to ask about a specific incident rather than the first one in context.diagnostics list. We could add better UX around this IMO by making which issue they are asking about clear in the code action name when it is added IF there are multiple konveyor diagnostics for a specific line. 


https://github.com/user-attachments/assets/e97e3410-e762-4bb5-85a4-2526cd47f4f1

Reply:
https://github.com/konveyor/editor-extensions/issues/476